### PR TITLE
(BSR) feat(agents): add test-dag-from-diff skill

### PR DESCRIPTION
**Pull request by dcuesta-pass**
## Summary
Adds a new Copilot agent skill `test-dag-from-diff` that automates the process of detecting which Airflow DAGs are affected by current git changes, creating test copies with the active branch as the default branch param, and uploading them to the target GCS Airflow bucket.

## Changes
- Add `.agents/skills/test-dag-from-diff/SKILL.md` with full instruction flow

## Related
- Complements the existing `data-pr-workflow` skill

## Testing
Skill manually validated on a feature branch — covers DAG detection, branch patching, GCS upload, and edge cases (master branch, no branch param, multiple DAG matches).

### CODE REVIEW

File: .agents/skills/test-dag-from-diff/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+Run these two commands to establish the working context:
+
+```bash
+git branch --show-current
+```
+
+Store the result as `CURRENT_BRANCH`.
+
+- If `CURRENT_BRANCH` is `master`, warn the user:
+  > ⚠️ You are on `master`. The branch param in the test DAG will point to `master`, which is the same as the production default. Consider running this from a feature branch to test your specific changes.
```
Comment (cibrahim-pass) [inline .agents/skills/test-dag-from-diff/SKILL.md:27]: should we explain that "production" here means stg not prod. Also, dev env is synced with the production branch not the master branch

File: .agents/skills/test-dag-from-diff/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+### 3. Map Changed Files to DAGs
+
+For each changed file path, determine a **search slug** using these rules:
+
+| Changed file pattern | Slug |
+|---|---|
+| `jobs/ml_jobs/<folder>/...` | `<folder>` |
+| `jobs/etl_jobs/external/<folder>/...` | `<folder>` |
+| `jobs/etl_jobs/internal/<folder>/...` | `<folder>` |
+| `jobs/etl_jobs/jobs/<folder>/...` | `<folder>` |
```
Comment (cibrahim-pass) [inline .agents/skills/test-dag-from-diff/SKILL.md:65]: shouldn't we only serach inside orchestration/dags ?

File: .agents/skills/test-dag-from-diff/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+Deduplicate the final list across all changed files.
+
+**If no DAGs are found**, inform the user and stop:
+> No associated DAGs found in `orchestration/dags/` for the changed files. You may need to deploy manually.
+
+---
+
+### 4. Confirm with User
+
+Present the matched DAGs clearly before doing anything:
```
Comment (cibrahim-pass) [inline .agents/skills/test-dag-from-diff/SKILL.md:95]: add for example so the agent knows the following bash is an example and would not deterministically respond with these two example dags (i think they are mostly smart to know that these are examples but shit happens)
